don't overwrite all environment variables

Smeb opened the pull request:
👋 hey, thanks for the plugin! We've found it very useful.

At the moment it has the somewhat tricky behavior of overriding the inner container's PATH and SHELL. We found this breaks openjdk images, for example. This code just blacklists environment variables which are already set by the image.

One downside of this would be if you foresee people intentionally overwriting the PATH in their drone secrets. If you think that's a problem we could try parsing to see if the variables changed.

rnorth commented:
Hi @Smeb, thanks!

It's interesting - we use this with all sorts of build images (including ones derived from adoptopenjdk) and haven't encountered this problem.

Still, I think it's sensible to do this, and I can't see huge downsides with blanket-blacklisting these particular values.

I'll just suggest one small change inline.

rnorth commented on command.sh:
```diff
@@ -52,7 +52,15 @@ docker pull ${PLUGIN_BUILD_IMAGE} 2>&1 | sed 's/^/   /g'
+  touch ${PWD}/outer_env_vars.env
+  case "$k" in
+      # avoid overwriting container's variables
+      DIND_COMMIT|DOCKER_CHANNEL|DOCKER_TLS_CERTDIR|DOCKER_VERSION|HOME|HOSTNAME|PATH|PWD|SHLVL)
```
```suggestion
      
# Note that the env vars to blacklist may be found using
#   `docker run -it --entrypoint env quay.io/testcontainers/dind-drone-plugin`
   DIND_COMMIT|DOCKER_CHANNEL|DOCKER_TLS_CERTDIR|DOCKER_VERSION|HOME|HOSTNAME|PATH|PWD|SHLVL)
```

rnorth replied:
I'm not sure if indentation will survive this patch, but I hope you get the gist.

Smeb commented:
I've added your patch and fixed the indentation, thanks for the quick response!

If you're interested, the specific image we're basing off of is:
`adoptopenjdk/openjdk8:jdk8u252-b09-debian-slim`, which has the following path by default (which we don't really change):
`PATH=/opt/java/openjdk/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin`

Smeb commented:
@rnorth - is there anything further I need to do on this PR before we can merge it?

rnorth reviewed (approved):
I'm very sorry for the long delay on merging, @Smeb. This looks good to me, so I'll merge and release.
